dali/kernels/tensor_shape.h

```diff
@@ -493,12 +493,12 @@ struct TensorListShape : TensorListShapeBase<TensorListShape<sample_ndim>, sampl
   TensorListShape(const std::vector<TensorShape<sample_ndim>> &sample_shapes)  // NOLINT
       : Base(flatten_shapes(sample_shapes)) {}
 
-  TensorListShape(const std::vector<int64_t> &shapes, int ndim = sample_ndim)  // NOLINT
+  TensorListShape(const std::vector<int64_t> &shapes, int ndim)
```

Comment:
Why? Isn't specifying the ndim both as template parameter and constructor argument to verbose?

Reply:
It was impossible to specify this vector using initializer list - the compiler got totally confused.